firmware/quadruna/VC/src/app/vehicle_dynamics/app_torqueVectoring.c

```diff
@@ -170,3 +121,163 @@ void app_torqueVectoring_handleAcceleration(void)
     app_canTx_VC_PIDPowerEstimateDerivative_set(pid_power_correction.derivative);
     app_canTx_VC_PIDPowerEstimateIntegral_set(pid_power_correction.integral);
 }
+
+/* Static functions*/
+
+static void app_torqueVectoring_stateMachine(float accelerator_pedal_percentage)
+{
+    // Read data from CAN
+    // NOTE: Pedal percent CAN is in range 0.0-100.0%
+    accelerator_pedal_percent   = accelerator_pedal_percentage;
+    wheel_speed_front_left_kph  = app_canRx_FSM_LeftWheelSpeed_get();
+    wheel_speed_front_right_kph = app_canRx_FSM_RightWheelSpeed_get();
+    motor_speed_left_rpm        = (float)app_canRx_INVL_MotorSpeed_get();
+    motor_speed_right_rpm       = -1 * (float)app_canRx_INVR_MotorSpeed_get();
+    battery_voltage             = app_canRx_BMS_TractiveSystemVoltage_get();
+    current_consumption         = app_canRx_BMS_TractiveSystemCurrent_get();
+    left_motor_temp_C           = app_canRx_INVL_MotorTemperature_get();
+    right_motor_temp_C          = app_canRx_INVR_MotorTemperature_get();
+    steering_angle_deg          = app_canRx_FSM_SteeringAngle_get();
+
+    // Power Limiting
+    power_limiting_inputs.left_motor_temp_C         = left_motor_temp_C;
+    power_limiting_inputs.right_motor_temp_C        = right_motor_temp_C;
+    power_limiting_inputs.accelerator_pedal_percent = accelerator_pedal_percent;
+    float estimated_power_limit;
+    estimated_power_limit = app_powerLimiting_computeMaxPower(&power_limiting_inputs);
+    // Power limit correction
+    float power_limit = estimated_power_limit * (1.0f + pid_power_correction_factor);
+
+    // initializing active diff outputs
+    active_differential_outputs.torque_left_Nm  = 0.0;
+    active_differential_outputs.torque_right_Nm = 0.0;
+
+    // initializing active diff outputs
+    traction_control_outputs.torque_left_Nm  = 0.0;
+    traction_control_outputs.torque_right_Nm = 0.0;
+
+    if (accelerator_pedal_percent <= PEDAL_NOT_PRESSED && torqueVectoringInternalState != OFF)
+    {
+        torqueVectoringInternalState = noPedalInput();
+        return;
+    }
+
+    switch (torqueVectoringInternalState)
+    {
+        case OFF:
+
+            if (accelerator_pedal_percent > 0.0f)
+            {
+                if (TRACTION_CONTROL_LOWER_BOUND <= steering_angle_deg &&
+                    steering_angle_deg <= TRACTION_CONTROL_UPPER_BOUND)
+                {
+                    torqueVectoringInternalState = runTractionControl(power_limit);
```

Comment:
based on the state and transition there could be different ways of dispatching a state call. please make those unifed.